src/ert/run_models/everest_run_model.py

```
@@ -787,24 +790,38 @@ def _forward_model_evaluator(
             else None
         )
 
-        # The simulation id's are a simple enumeration over the evaluated
-        # forward models. For the evaluated controls they are therefore
-        # implicitly given by there position in the evaluated_control_indices
-        # list. We store for each control vector that id, or -1 if it was not
-        # evaluated:
-        sim_ids = np.array(
-            [
-                ei.simulation_id if ei.simulation_id is not None else -1
-                for ei in evaluation_infos
-            ],
-            dtype=np.int32,
-        )
+        # We also return for each evaluated control vector the corresponding
+        # simulation id. If the results came from the cache, we get the
+        # simulation id from the source batch. We aso store the source batch of
+        # the results, from from the cache info if the results was cached; if
+        # not the current batch is stored as the source. Results that were not
+        # active and therefore were not evaluated receive -1 values.
+        sim_ids = []
+        source_batch_ids = []
+        for ei in evaluation_infos:
+            if ei.simulation_id is None:
+                sim_id, batch_id = next(
```

If it is not in the cache hit dict i would raise an error, if it has the status `CACHED` it should always be there I think. So either raise an explicit error or remove the silent failure defaulting to (-1, -1)

---

There is also the case that a vector of controls is passed, but marked as inactive. In that case it is not evaluated at all. So there is nothing run or cached, and we need a place-holder.

---

Forgot about the inactive ones, would prefer adding another if checking if they are flagged as `INACTIVE` instead of defaulting, but I see we do need to indicate the inactive-ness somehow now